File: ClientApp/src/view-models/user-view-model.ts
import { AgeGroupViewModel } from "./age-group-view-model";
import { EducationViewModel } from "./education-view-model";
import { GenderViewModel } from "./gender-view-model";
import { LanguageViewModel } from "./language-view-model";
import { PositionViewModel } from "./position-view-model";
import { RegionViewModel } from "./region-view-model";
import { SectorOfActivityViewModel } from "./sector-of-activity-view-model";

export class UserViewModel {
  public id: string;
  public firstName: string;
  public lastName: string;
  public email: string;
  public age: number;
  public gender: GenderViewModel;
  public regions: Array<RegionViewModel>;
  public educations: Array<EducationViewModel>;
  public sectorsOfActivities: Array<SectorOfActivityViewModel>;
  public professionalEmail: string;
  public bio: string;
  public languages: Array<LanguageViewModel>;
  public website: string;
  public myerBriggsCode: string;
  public accessToken: string;
  public refreshToken: string;
  public positions: Array<PositionViewModel>;
  public profileImageName: string;
  public phoneNumber: string;
  public ageGroup: AgeGroupViewModel;
}

File: ViewModels/UserViewModel.cs
using KPProject.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace KPProject.ViewModels
{
    public class UserViewModel
    {
        public string Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int Age { get; set; }
        public Gender Gender { get; set; }
        public string Email { get; set; }
        public List<RegionModel> Regions { get; set; }
        public List<PositionModel> Positions { get; set; }
        public List<EducationModel> Educations { get; set; }
        public List<SectorOfActivityModel> SectorsOfActivities { get; set; }
        public AgeGroupModel AgeGroup { get; set; }
        public string PhoneNumber { get; set; }
        public string ProfessionalEmail { get; set; }
        public string Bio { get; set; }
        public List<LanguageModel> Languages { get; set; }
        public string Website { get; set; }
        public string MyerBriggsCode { get; set; }
        public string AccessToken { get; set; }
        public string RefreshToken { get; set; }
        public string ProfileImageName { get; set; }
    }
}
